test/test_constraints_file.py

```diff
@@ -50,18 +50,100 @@ def test_add_at_least_one_employee_per_shift_constraint():
 
 
 # A Test to check if there is an optimal solution with 0 employees, while satisfying the
-# 'add_at_least_one_employee_per_shift_constraint' constraint.
+# 'add_exactly_one_employee_per_shift_constraint' constraint.
 # Expected result is not an Optimal solution because there is no employee to assign to the shift.
-def test_add_at_least_one_employee_per_shift_constraint_with_no_employees():
+def test_add_exactly_one_employee_per_shift_constraint_with_no_employees():
     employees = []
 
     test_shift = Shift(shift_id=uuid4(), shift_type=ShiftTypesEnum.MORNING, start_time=datetime.datetime(2023, 12, 11, 9, 30), end_time=datetime.datetime(2023, 12, 11, 16, 0))
     model = cp_model.CpModel()
 
     shifts = generate_shift_employee_combinations(employees, [test_shift], model)
-    add_at_least_one_employee_per_shift_constraint([test_shift], employees, model, shifts)
+    add_exactly_one_employee_per_shift_constraint([test_shift], employees, model, shifts)
 
     solver = cp_model.CpSolver()
     status = solver.Solve(model)
 
     assert (status != cp_model.OPTIMAL)
+
+
+# A Test to check if there is an optimal solution with 2 employees, while satisfying the
+# 'add_at_most_one_shift_in_the_same_day_constraint' constraint.
+# Expected result is an Optimal solution because there is a solution where no employee is assigned to shifts.
+def test_add_at_most_one_shift_per_employee_in_the_same_day_constraint_without_add_exactly_one_employee_per_shift_constraint():
```

Comment:
this is confusing. write a sentence

Reply:
also why is noone doing any shift an optimal solution? didnt we say all shifts need an employee doing them?